doc: add missing documentation in PR76408

valeriosetti opened the pull request:
PR #76408 introduced some kconfig changes (some symbols were renamed while a new one was added), but none of these were documented. This PR covers this gap amending both migration guide and release notes.

tomi-font commented on doc/releases/migration-guide-4.0.rst:
```diff
@@ -40,6 +40,9 @@ Mbed TLS
 
 * The Kconfig options ``CONFIG_MBEDTLS_TLS_VERSION_1_0`` and ``CONFIG_MBEDTLS_TLS_VERSION_1_1``
   have been removed because Mbed TLS doesn't support TLS 1.0 and 1.1 anymore since v3.0. (:github:`76833`)
+* Following Kconfig symbols were renamed (:github:`76408`):
+  * ``CONFIG_MBEDTLS_ENTROPY_ENABLED`` is now ``CONFIG_MBEDTLS_ENTROPY_C``,
```
```suggestion
* The following Kconfig symbols were renamed (:github:`76408`):
  * ``CONFIG_MBEDTLS_ENTROPY_ENABLED`` is now :kconfig:option:`CONFIG_MBEDTLS_ENTROPY_C`,
```

tomi-font commented on doc/releases/release-notes-4.0.rst:
```diff
@@ -315,6 +315,11 @@ Libraries / Subsystems
   * Mbed TLS was updated to version 3.6.1. The release notes can be found at:
     https://github.com/Mbed-TLS/mbedtls/releases/tag/mbedtls-3.6.1
+  * New Kconfig symbol :kconfig:option:`CONFIG_MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG_ALLOW_NON_CSPRNG`
+    was added to allow ``psa_get_random()`` to make use of non cryptographically
```
```suggestion
    was added to allow ``psa_get_random()`` to make use of non-cryptographically
```

tomi-font commented on doc/releases/release-notes-4.0.rst:
```diff
@@ -315,6 +315,11 @@ Libraries / Subsystems
+  * New Kconfig symbol :kconfig:option:`CONFIG_MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG_ALLOW_NON_CSPRNG`
+    was added to allow ``psa_get_random()`` to make use of non cryptographically
+    secure random sources when :kconfig:option:`CONFIG_MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG`
+    is also set. This is only meant to be used for test purposes, not in production.
```
```suggestion
    is also enabled. This is only meant to be used for test purposes, not in production.
```

tomi-font commented on doc/releases/release-notes-4.0.rst:
```diff
@@ -315,6 +315,11 @@ Libraries / Subsystems
 
   * Mbed TLS was updated to version 3.6.1. The release notes can be found at:
     https://github.com/Mbed-TLS/mbedtls/releases/tag/mbedtls-3.6.1
+  * New Kconfig symbol :kconfig:option:`CONFIG_MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG_ALLOW_NON_CSPRNG`
```
```suggestion
  * A new Kconfig symbol (:kconfig:option:`CONFIG_MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG_ALLOW_NON_CSPRNG`)
```
or
```suggestion
  * The Kconfig symbol :kconfig:option:`CONFIG_MBEDTLS_PSA_CRYPTO_EXTERNAL_RNG_ALLOW_NON_CSPRNG`
```

tomi-font reviewed (changes requested):
Great!

mmahadevan108 reviewed (approved):
Can the title of the commit be changed to mention about the new Kconfig added instead of stating it is adding missing documentation for a PR.

mmahadevan108 reviewed (changes requested):
Request to update the commit title

valeriosetti committed:
doc: rename Mbed TLS entropy Kconfigs

This change was done in #76408, but without documenting it.
This commit adds missing documentation.

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

valeriosetti committed:
doc: new Kconfig to allow non-CS sources in psa_generate_random()

This change was done in #76408, but without documenting it.
This commit adds missing documentation.

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

valeriosetti commented:
> Can the title of the commit be changed to mention about the new Kconfig added instead of stating it is adding missing documentation for a PR.

Sure, I just did this. I split the single commit into 2 because it was easier to add the proper commits' title then.

kartben commented on doc/releases/migration-guide-4.0.rst:
```diff
@@ -40,6 +40,9 @@ Mbed TLS
 * The Kconfig options ``CONFIG_MBEDTLS_TLS_VERSION_1_0`` and ``CONFIG_MBEDTLS_TLS_VERSION_1_1``
   have been removed because Mbed TLS doesn't support TLS 1.0 and 1.1 anymore since v3.0. (:github:`76833`)
+* The following Kconfig symbols were renamed (:github:`76408`):
+  * ``CONFIG_MBEDTLS_ENTROPY_ENABLED`` is now :kconfig:option:``CONFIG_MBEDTLS_ENTROPY_C``,
+  * ``CONFIG_MBEDTLS_ZEPHYR_ENTROPY`` is now :kconfig:option:``CONFIG_MBEDTLS_ENTROPY_POLL_ZEPHYR``.
```
I'll fix this as part of 78266 but note that the kconfig:option: role calls for single backticks, not double. Luckily, these kind of easy-to-make mistakes will soon be caught by sphinx-lint!
